server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DomainMappingsRoutesTest.java

```diff
@@ -0,0 +1,217 @@
+
+        @Test
+        void addDomainMappingShouldRespondWithNoContent() {
+            // @formatter:off
+            given(with().body("to.com"))
+                        .put("from.com")
+                    .then()
+                        .statusCode(HttpStatus.NO_CONTENT_204)
+                        .body(isEmptyString());
+            // @formatter:on
+        }
+
+        @Test
+        void getDomainMappings() throws RecipientRewriteTableException {
+            Domain expectedDomain = Domain.of("abc.com");
+            MappingSource mappingSource = MappingSource.fromDomain(expectedDomain);
+            ImmutableList<String> expectedAliases = ImmutableList.of("to_1.com", "to_2.com", "to_3.com");
+
+            for (String alias : expectedAliases) {
+                recipientRewriteTable.addAliasDomainMapping(mappingSource, Domain.of(alias));
+            }
+
+            // @formatter:off
+            final Map<String, List<String>> map =
+                    when()
+                        .get()
+                    .then()
+                        .contentType(ContentType.JSON)
+                        .statusCode(HttpStatus.OK_200)
+                    .extract()
+                        .body()
+                        .jsonPath()
+                        .getMap(".");
+            // @formatter:on
+
+            assertThat(map)
+                    .containsOnly(entry(expectedDomain.name(), expectedAliases));
+        }
+
+        @Test
+        void getDomainMappingsShouldBeEmpty() {
+            // @formatter:off
+            when()
+                .get()
+            .then()
+                .contentType(ContentType.JSON)
+                .statusCode(HttpStatus.OK_200)
+                .body(is("{}"));
+            // @formatter:on
+        }
+
+        @Test
+        void deleteDomainMappingShouldRespondWithNoContent() {
+            // @formatter:off
+            given(with().body("to.com"))
+                        .delete("from.com")
+                    .then()
+                        .statusCode(HttpStatus.NO_CONTENT_204)
+                        .body(isEmptyString());
+            // @formatter:on
+        }
+    }
+
+    @Nested
+    class IllegalInputs {
+        @Test
+        void addDomainMappingShouldRespondWithNotFound() {
+            // @formatter:off
+            when()
+                .put("")
+            .then()
+                .statusCode(HttpStatus.NOT_FOUND_404);
+            // @formatter:on
+        }
+
+        @Test
+        void deleteDomainMappingShouldRespondWithNotFound() {
+            // @formatter:off
+            when()
+                .delete("")
+            .then()
+                .statusCode(HttpStatus.NOT_FOUND_404);
+            // @formatter:on
+        }
+
+        @ParameterizedTest
+        @MethodSource("org.apache.james.webadmin.routes.DomainMappingsRoutesTest#invalidInputs")
+        void addDomainMappingWithInvalidDomainInPath(String fromDomain, String toDomain) {
```

Comment:
I think that parametrized test is a great idea and thank you for sharing it with us, that is really instructive.

However, you should know that we recently encountered issues with junit 5.2.0, namely:
 - tests not played, missing dependencies, etc...

As such I'd like to avoid using it until it get more mature and that the maven integration improves.

Furthermore, I'd like to avoid diverging junit version between projects. 

Here, we got 2 inputs... I'd propose you to flatten them and write one test for each invalid case, without parametrized tests (for the reasons I listed above).